icu4c/source/i18n/number_fluent.cpp

```diff
@@ -288,6 +288,26 @@ Derived NumberFormatterSettings<Derived>::usage(const StringPiece usage)&& {
     return move;
 }
 
+template <typename Derived>
+Derived NumberFormatterSettings<Derived>::displayOptions(const DisplayOptions &displayOptions) const & {
+    Derived copy(*this);
+    //`displayCase` does not recognise the `undefined`, therefore, we use get grammatical case with null
+    // function.
```

Comment:
Why not make fMacros.unitDisplayCase() understand "undefined"?

Reply:
after removing `unitDiplayCase`, we need to remove also from macros unitDisplayCase and replace it with `GrammaticalCase` or `DisplayOptions. For now, and because we need to have unitDisplayCase for ICU72 (as deprecated). Therefore, we are using it.